### PR TITLE
Ticket31 update db

### PR DESCRIPTION
**Pull request by sergiomarrocoli**
Don't merge until after ticket-10 has been merged 

- Restructures the db and classes to match form requirements, like changing some fields to asssociations (e.g. 'link') and belongs_to to has_many
- Rake tasks to migrate the data to this new structure (e.g. adds the old commitment.country to commitment.countries)
- updates existing importer task to match new data structure

## clear your local db:
bundle exec rake db:drop
bundle exec rake db:create

### download the staging db (you might need to add credentials):
aws s3 ls s3://actionagenda.bkp --recursive --human-readable --summarize
aws s3 cp s3://actionagenda.bkp/staging-db/actiona_agenda_staging.tar.gz action_agenda_copy.tar

### Unzip the file
however you want

### import the data to your db:
psql -U :your_user -d aa_development -f actionagenda_staging.sql

### check these run without errors:
bundle exec rake db:migrate

bundle exec rake migrate_data:add_commitment_country_to_commitment_countries
bundle exec rake migrate_data:add_commitment_link_to_commitment_links
bundle exec rake migrate_data:add_default_options_for_threats
bundle exec rake migrate_data:set_existing_commitments_to_live

### check the site is still working ok
restart webacker and server, click around

I would also reset your db in the normal way, run seeds and the importer to check that still works

### CODE REVIEW

File: lib/tasks/migrate_data.rake [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    puts "default options added"
+  end
+
+  task set_existing_commitments_to_live: :environment do
+    failed_to_set_live = []
+    Commitment.find_each do |commitment|
+      commitment.state = :live
+      failed_to_set_live << commitment.id unless commitment.save
+    end
+    puts failed_to_set_live.empty? ? "All commitments set to 'live'." : "Commitements #{ failed_to_set_live.join(', ')} could not be set to live"
```
Comment (yndajas) [inline lib/tasks/migrate_data.rake:34]: Typo in end of ternary statement: "Commitements"

File: test/fixtures/commitments.yml [hunk 2]
```diff
@@ -1,33 +1,50 @@
...
   implementation_year: 1
-  update_year: 1
   name: MyText
   governance_authority: MyText
   planned_actions: MyText
   description: MyText
-  link: MyString
+  managers: indigenous, custom
+  state: live
+  stage: 'Committed'
```
Comment (yndajas) [inline test/fixtures/commitments.yml:18]: Mixture of single quotes and no quotes

**Review by yndajas — CHANGES_REQUESTED** [review]
Just a few typos and format comments, plus a couple of queries on associations

File: app/models/link.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Link < ApplicationRecord
+  has_and_belongs_to_many :commitments
```
[the remark this answers is not in the document]
Comment (sergiomarrocoli) [reply app/models/link.rb:2]: you're right, should just be has_many and belongs_to

**Review by yndajas — APPROVED** [review]

**Pull request merged**